DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -162,8 +162,24 @@ internal async Task<IReadOnlyList<TransportMember>> SearchMembersAsync(ulong gui
             var url = Utilities.GetApiUriFor(path, BuildQueryString(querydict));
             var res = await this.DoRequestAsync(this.Discord, bucket, url, RestRequestMethod.GET, route).ConfigureAwait(false);
             var json = JArray.Parse(res.Response);
+            var tms = json.ToObject<IReadOnlyList<TransportMember>>();
+            foreach (var xtm in tms)
+            {
+                var usr = new DiscordUser(xtm.User) { Discord = this.Discord };
+
+                this.Discord.UserCache.AddOrUpdate(xtm.User.Id, usr, (id, old) =>
+                {
+                    old.Username = usr.Username;
+                    old.Discord = usr.Discord;
+                    old.AvatarHash = usr.AvatarHash;
+
+                    return old;
+                });
+            }
+
+            var mbrs = tms.Select(tm => new DiscordMember(tm) { Discord = this.Discord, _guild_id = guild_id });
```

Comment:
Just a nit-pick, but instead of enumerating over this twice, you could create the mbrs list outside the loop, and add to that whilst updating the user cache